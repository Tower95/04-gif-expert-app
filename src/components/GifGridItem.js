import React from 'react';
import PropsTypes from 'prop-types';

const GifGridItem = ({ title, url }) => {
  return (
    <div className="card animate__animated  animate__fadeInRight">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
}

export default GifGridItem;

GifGridItem.propTypes = {
  title: PropsTypes.string.isRequired,
  url: PropsTypes.string.isRequired
}