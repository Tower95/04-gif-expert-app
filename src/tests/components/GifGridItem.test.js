import React from "react";
import { shallow } from 'enzyme'

import GifGridItem from "../../components/GifGridItem";

describe('prubas en <GifGridItem', () => {

  const title = 'Dragon Ball'
  const url = 'http://localhost/algo.jpg'

  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('deve de mostrar <GifGridItem /> correctamente', () => {

    expect(wrapper).toMatchSnapshot();

  })

  test('debe de tener un parrafo con el title', () => {
    const p = wrapper.find('p');

    expect(p.text().trim()).toBe(title);

  })

  test('debe tener la imagen con url y alt', () => {

    const img = wrapper.find('img');

    console.log(img.html());

    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);


  })

  test('div debe de tener la clase animate__fadeInRight', () => {

    const div = wrapper.find('div');

    expect(div.hasClass('animate__fadeInRight')).toBe(true);

  })

})
