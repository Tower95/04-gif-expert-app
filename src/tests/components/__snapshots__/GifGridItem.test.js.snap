// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prubas en <GifGridItem deve de mostrar <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated  animate__fadeInRight"
>
  <img
    alt="Dragon Ball"
    src="http://localhost/algo.jpg"
  />
  <p>
    Dragon Ball
  </p>
</div>
`;
